packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -928,6 +930,32 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
     return selection.copyWith(extentOffset: nextExtent);
   }
 
+  // Later to use for a more accurate drag-to-extend-selection implementation
+  // In place of selectParagraphAt
+  TextSelection? _dragStartInitialSelection;
```

Comment:
I'm not using this yet and it's pretty rough still. I have played around with using these functions in `onDragSelectionStart` and `onDragSelectionUpdate` to provide a more accurate text selection behavior.